aten/src/ATen/core/ivalue.cpp

```diff
@@ -630,6 +631,17 @@ getClassConverter() {
   return classConverter;
 }
 
+template <typename T>
```

Comment:
I'm not sure if this is the best way to approach this, but the reason this function needs to be templated is because it needs to propagate the value of `callback()`. Normally the template would be defined in the header to avoid the explicit instantiation below, but there were some compile errors when adding the necessary `include <ATen/ThreadLocalState.h>` to `ivalue_inl.h` (it looks like it caused a circular header issue).